packages/connections/src/stores/connections-store.spec.ts

```diff
@@ -0,0 +1,225 @@
+      loadAll: loadAllSpy,
+      save: saveSpy,
+    };
+  });
+
+  describe('#loadConnections', function () {
+    it('loads the connections from the connection storage', async function () {
+      const loadAllSpyWithData = sinon.fake.resolves(mockConnections);
+      mockConnectionStorage.loadAll = loadAllSpyWithData;
+
+      const { result } = renderHook(() =>
+        useConnections(noop, mockConnectionStorage, noop)
+      );
+
+      // Wait for the async loading of connections to complete.
+      await waitFor(() =>
+        expect(result.current[0].connections.length).to.equal(2)
+      );
+
+      expect(loadAllSpyWithData.callCount).to.equal(1);
+      expect(result.current[0].connections.length).to.equal(2);
+    });
+  });
+
+  describe('#saveConnection', function () {
+    describe('with an existing connection', function () {
+      let hookResult: RenderResult<ReturnType<typeof useConnections>>;
+      beforeEach(async function () {
+        mockConnectionStorage.loadAll = () => Promise.resolve(mockConnections);
+
+        const { result } = renderHook(() =>
+          useConnections(noop, mockConnectionStorage, noop)
+        );
+
+        // Wait for the async loading of connections to complete.
+        await waitFor(() =>
+          expect(result.current[0].connections.length).to.equal(2)
+        );
+
+        await act(async () => {
+          await result.current[1].saveConnection({
+            id: 'oranges',
+            connectionOptions: {
+              connectionString: 'aba',
+            },
+            favorite: {
+              name: 'not peaches',
+            },
+          });
+        });
+
+        hookResult = result;
+      });
+
+      it('calls to save a connection on the store', function () {
+        expect(saveSpy.callCount).to.equal(1);
+      });
+
+      it('updates the existing entry on the connections list', function () {
+        expect(hookResult.current[0].connections.length).to.equal(2);
+        expect(hookResult.current[0].connections[1]).to.deep.equal({
+          id: 'oranges',
+          connectionOptions: {
+            connectionString: 'aba',
+          },
+          favorite: {
+            name: 'not peaches',
+          },
+        });
+      });
+
+      it('clones the existing connection when it is updated', function () {
+        expect(hookResult.current[0].connections[1]).to.not.equal(
+          mockConnections[1]
+        );
+        expect(
+          hookResult.current[0].connections[1].connectionOptions
+        ).to.not.equal(mockConnections[1].connectionOptions);
+      });
+    });
+
+    describe('saving a new connection', function () {
+      let hookResult: RenderResult<ReturnType<typeof useConnections>>;
+      beforeEach(async function () {
+        const { result } = renderHook(() =>
+          useConnections(noop, mockConnectionStorage, noop)
+        );
+
+        await act(async () => {
+          await result.current[1].saveConnection({
```

Comment:
First time using this `@testing-library/react-hooks` library but seems pretty neat for testing some more complicated hooks (hopefully not often). cc @gribnoysup